File: Programme/WMS/WMS_CL/Lager/Produkt.cs
public class Produkt
{
    private string _myID = "P";
    public Produkt () { }
    public Produkt (string PID, string Name, int Menge, int PreisProEinheit, Ebene MeineEbene)
    {
        _myID = PID;
        this.Name = Name;
        this.Menge = Menge;
        this.PreisProEinheit = PreisProEinheit;
        this.MeineEbene = MeineEbene;
    }

    public Produkt(int PID, string Name, int Menge, int PreisProEinheit, Ebene MeineEbene)
    {
        _myID += PID;
        this.Name = Name;
        this.Menge = Menge;
        this.PreisProEinheit = PreisProEinheit;
        this.MeineEbene = MeineEbene;
    }

    public string PID
	{
		get
        {
            return MeineEbene.EID + "-" + _myID;
        }
        set { _myID = value; }
	}

	public string Name
	{
		get;
		set;
	}

	public int Menge
	{
		get;
		set;
	}

	public int PreisProEinheit
	{
		get;
		set;
	}

    public Ebene MeineEbene
    {
        get;
        set;
    }

    public override string ToString()
    {
        return Name + " " + Menge + " Stück";
    }
}

File: Programme/WMS_Java/WMS_Projekt/WarehouseManagementSystem/src/data/Produkt.java
package data;

public class Produkt {

    private String pId;
    private String name;
    private int menge;
    private double einheitspreis;
    private String ebenenId;



    public Produkt(String id, String name, int menge, double preis, String eId) {
        setpId(id);
        setName(name);
        setMenge(menge);
        setEinheitspreis(preis);
        setEbenenId(eId);
    }

    public String getpId() {
        return pId;
    }
    private void setpId(String pId) {
        this.pId = pId;
    }

    public String getName() {
        return name;
    }
    private void setName(String name) {
        this.name = name;
    }

    public int getMenge() {
        return menge;
    }
    private void setMenge(int menge) {
        this.menge = menge;
    }

    public double getEinheitspreis() {
        return einheitspreis;
    }
    private void setEinheitspreis(double einheitspreis) {
        this.einheitspreis = einheitspreis;
    }

    public String getEbenenId() {
        return ebenenId;
    }
    private void setEbenenId(String ebenenId) {
        this.ebenenId = ebenenId;
    }
}
